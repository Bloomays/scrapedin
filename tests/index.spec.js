const fs = require("fs");
const util = require('util');
const scrapedin = require('../src/scrapedin');

let cookies;

beforeAll(() => {
    try {
        cookies = fs.readFileSync(__dirname+'/cookie.json');
    }
    catch(error) {
        console.warn('unable to load cookie.json', error);
    }
    
})

const urls = [
    ['https://www.linkedin.com/in/lina-morel-9b1b7b1b8/', 'lina', 0],
    ['https://www.linkedin.com/in/rosa-barbet-69a6bb220/', 'rosa', 0],
    ['https://www.linkedin.com/in/aurore-veneto-5a6034213/', 'aurore', 0]
];

test.each(urls)("should get profiles properly", async(url, name, nbReco) => {
    let profile = {};
    try {
        if (!cookies) {
            console.warn('missing cookie.json, must exit');
            return;
        }
        const options = {
            cookies: JSON.parse(cookies),
            isHeadless: true,
            hasToLog: true,
            hasToGetContactInfo: true
        };
        const profileScraper = await scrapedin(options);
        profile = await profileScraper(url);
        // console.log(util.inspect(profile, false, null, true));
        fs.writeFileSync(__dirname+'/profiles/' + name + '.json', JSON.stringify(profile, null, 4));
        if (profile?.recommendations?.given) {
            profile.recommendations.given = profile.recommendations.given.map((reco) => {
                if (reco.profileImage && reco.profileImage.indexOf('https://') === 0) {
                    reco.profileImage = 'https://good.com'
                }
                return reco;
            });
        }
    }
    catch (error) {
        if (error.image) {
            fs.writeFile(__dirname + '/profiles/error.' + name + '.png', error.image, 'base64', function(err) {
                console.log(err);
            });
        }
        console.error(error.message);
    }
    finally {
        expect(profile).toMatchSnapshot({
            profile: {
                imageurl: expect.any(String)
            }
        }, name);
    }
}, 10000000)