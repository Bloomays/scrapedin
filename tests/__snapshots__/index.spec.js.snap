// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get profiles properly 1`] = `
{
  "about": {
    "text": "Ruby on Rails / Node.js developer with a knack for bleeding edge technologies. Specialties: front/back end web development.",
  },
  "accomplishments": [],
  "contact": [
    {
      "links": [
        "https://www.linkedin.com/in/bertrand-chardon-5876aab",
      ],
      "type": "Profil de Bertrand",
      "values": [
        "linkedin.com/in/bertrand-chardon-5876aab",
      ],
    },
    {
      "links": [
        "http://www.smartdate.com/",
      ],
      "type": "Site web",
      "values": [
        "smartdate.com (Entreprise)",
      ],
    },
  ],
  "courses": [],
  "educations": [
    {
      "date1": "1998",
      "date2": "2003",
      "degree": "Master, Computer Science",
      "title": "Universit√© Paris-Sud",
      "url": "https://www.linkedin.com/company/1165891/",
    },
  ],
  "languages": [
    {
      "name": "English",
      "proficiency": "Bilingue ou langue natale",
    },
    {
      "name": "Spanish",
      "proficiency": "Comp√©tence professionnelle",
    },
  ],
  "peopleAlsoViewed": [],
  "positions": [
    {
      "companyName": "Doctrine",
      "companyNameWithRole": undefined,
      "link": "https://www.linkedin.com/company/10511637/",
      "roles": [
        {
          "date1": "sept. 2021",
          "date2": "aujourd‚Äôhui",
          "location": "Paris, √éle-de-France, France",
          "title": "Data Engineer",
        },
        {
          "date1": "juin 2019",
          "date2": "aujourd‚Äôhui",
          "description": "Dites adieu aux doutes. 

Construisez vos strat√©gies et conseils juridiques sur des certitudes. 

Pl√©biscit√©e par pr√®s de 7 000 avocats et juristes, Doctrine est la 1re plateforme d'intelligence juridique qui rend votre analyse juridique plus fiable gr√¢ce √† la technologie.

üëâüèº D√©couvrez pourquoi ils nous ont choisi sur https://www.doctrine.fr/avis 

Une innovation unique en Europe : avec le Document Analyzer, v√©rifiez la validit√© de l‚Äôargumentaire juridique de la partie adverse et consolidez le v√¥tre. Economisez des heures de recherche en acc√©dant en un clic aux sources juridiques cit√©es dans des √©critures.

Essayez-le gratuitement ici : https://www.doctrine.fr/document-analyzer

üíö Rejoignez l'√©quipe https://www.welcometothejungle.com/fr/companies/doctrine/jobs

üö® Retrouvez les derni√®res nouveaut√©s de la plateforme : https://nouveautes.doctrine.fr",
          "location": "Paris Area, France",
          "title": "Backend Developer",
        },
      ],
      "url": "https://www.linkedin.com/company/10511637/",
    },
    {
      "companyName": "Orson | Website Builder",
      "date1": "mai 2013",
      "date2": "avr. 2019",
      "description": "My role at Orson encompasses various missions:

- lead developer (developing features in close connection with the CTO/founder)
- assisting with the integration of new hirees through documentation, guidelines and mentoring
- providing technical solutions in close collaboration with the CTO to best incorporate new features into the product
- ensuring the stability and maintenance of the platform
- working in close proximity with the retention team to ensure that user feedback is taken into account on a daily basis
- constant technology watch and monitoring to ensure that the tools are always a good fit to us and the market
- hardening the development and deployment processes to eliminate downtime and pitfalls
- providing general directions in terms of product development from a technical standpoint (resource usage, usability on all platforms, using the right tool for the right job)

The development cycle usually takes the form of runs where a given feature is first tested through prototypes which are consequently iteratied upon (which will usually entail technical discussions on the merits of the various possible solutions, measuring). 
Said runs are usually in the order of magnitude of a few weeks, to ensure agility and an adaptive development cycle.

The thinking is, as a result, that of an upstream to downstream approach where product, implementation and testing are all integrated into the discussion with the different people involved.",
      "link": "https://www.linkedin.com/company/10866480/",
      "location": "Paris Area, France",
      "roles": [],
      "title": "D√©veloppeur Ruby on Rails",
      "url": "https://www.linkedin.com/company/10866480/",
    },
    {
      "companyName": "RescuedCode",
      "date1": "ao√ªt 2011",
      "date2": "juil. 2013",
      "description": "Working on wonderful technology (Rails 3 / Backbone goodness) with gifted developers.",
      "link": "https://www.linkedin.com/search/results/all/?keywords=RescuedCode",
      "location": "Vancouver, Canada",
      "roles": [],
      "title": "Contractor",
      "url": "https://www.linkedin.com/search/results/all/?keywords=RescuedCode",
    },
    {
      "companyName": "Smartdate",
      "date1": "mai 2010",
      "date2": "mai 2011",
      "description": "Rails developer, added tests, features, tools for the team (managing i18n translations, user profile moderation).",
      "link": "https://www.linkedin.com/company/713199/",
      "roles": [],
      "title": "Web Developer",
      "url": "https://www.linkedin.com/company/713199/",
    },
    {
      "companyName": "AdPerfect",
      "date1": "janv. 2009",
      "date2": "juil. 2009",
      "description": "Developping ad loaders using the Javascript DOM API and jQuery/SWFObject. Interfacing with talented Flash/ActionScript developers to inject rich content classifieds on target web pages.",
      "link": "https://www.linkedin.com/company/110365/",
      "location": "New Westminster, British Columbia, Canada",
      "roles": [],
      "title": "Client-side Javascript Developer",
      "url": "https://www.linkedin.com/company/110365/",
    },
    {
      "companyName": "INRIA Futurs, LRI, Univ. Paris-Sud",
      "date1": "2005",
      "date2": "2008",
      "description": "Incrementing and improving a web-based scientific conference management system coded with PHP/MySQL/Javascript.
Involved adding features, testing, fixing bugs and interfacing with a researcher to gather business knowledge about conferences.",
      "link": "https://www.linkedin.com/search/results/all/?keywords=INRIA+Futurs%2C+LRI%2C+Univ%2E+Paris-Sud",
      "roles": [],
      "title": "Ing√©nieur Expert",
      "url": "https://www.linkedin.com/search/results/all/?keywords=INRIA+Futurs%2C+LRI%2C+Univ%2E+Paris-Sud",
    },
  ],
  "profile": {
    "connections": "264¬†relations",
    "headline": "Data Engineer at Doctrine",
    "imageurl": Any<String>,
    "location": "Paris et p√©riph√©rie",
    "name": "Bertrand Chardon",
    "summary": "Ruby on Rails / Node.js developer with a knack for bleeding edge technologies. Specialties: front/back end web development.",
  },
  "projects": [],
  "recommendations": {
    "given": [
      {
        "profileImage": "https://good.com",
        "text": "I had the chance to work with Nilo while developing a dating website in Paris, he was the product owner at the time. His natural empathy and charism as well as his natural North-American efficiency-centered work ethics were an inspiration for everyone.  His greatest strength is his natural ability to reach that perfect balance between smooth human rapport and a result-driven business culture.  It should also be noted that Nilo helped several of his colleagues grow in their respective roles at Smartdate at the time, effectively (and sucessfully) playing the role of a mentor to some of his younger coworkers.  I would highly recommend Nilo to any company.",
        "userDescription": "Technology | Health Care | Innovation | Helping organizations build digital products",
      },
      {
        "text": "With youth usually comes an uncanny ability to think too big, to seek unneeded elegance and to run astray from the path of highest efficiency.  In the 8 months I spent at Smartdate working with him, Damien showed a very strong motivation, was able to seek counselling and follow advice, never let his ego take the best out of projects.  He demonstrated a very good ability to improve and learn and his immersion in North-American culture was definitely a key factor that helped him succeed in an international team.  I would gladly recommend him to any potential recruiter.",
        "userDescription": "Cyber security specialist",
      },
      {
        "text": "Howard is an extremely focused and thorough developer. On top of his evident skills, he knows how to make people working around him improve and develop their programming toolset through advice, examples and guidance. He is the type of developer that would continuously look for new ways and techniques to make his code better and is well-versed in all facets of programming, be it designing, coding, fine-tuning applications with performance in mind or testing.  His main quality is his ability to assess a situation and approach it in a scientific manner, looking at pure facts and basing his decisions and choices upon that analysis.  I would highly recommend Howard to any potential employer or colleague.",
        "userDescription": "Head of Backend Engineering at Getaround",
      },
    ],
    "givenCount": "0",
    "received": [],
    "receivedCount": "0",
  },
  "skills": [
    {
      "title": "Ruby on Rails",
    },
    {
      "title": "MySQL",
    },
    {
      "title": "HAML",
    },
    {
      "title": "JavaScript",
    },
    {
      "title": "Web Development",
    },
    {
      "title": "Git",
    },
    {
      "title": "Ruby",
    },
    {
      "title": "HTML5",
    },
    {
      "title": "jQuery",
    },
    {
      "title": "Web Applications",
    },
    {
      "title": "Scrum",
    },
    {
      "title": "Agile",
    },
    {
      "title": "XML",
    },
    {
      "title": "JSON",
    },
    {
      "title": "SQL",
    },
    {
      "title": "AJAX",
    },
    {
      "title": "REST",
    },
    {
      "title": "HTML 5",
    },
    {
      "title": "Backbone.js",
    },
    {
      "title": "Test Driven Development",
    },
    {
      "title": "HAML",
    },
    {
      "title": "Web Development",
    },
    {
      "title": "Web Applications",
    },
    {
      "title": "Scrum",
    },
    {
      "title": "Agile",
    },
    {
      "title": "Test Driven Development",
    },
    {
      "title": "Agile Methodologies",
    },
    {
      "title": "Software Development",
    },
    {
      "title": "Ruby on Rails",
    },
    {
      "title": "MySQL",
    },
    {
      "title": "JavaScript",
    },
    {
      "title": "Git",
    },
    {
      "title": "Ruby",
    },
    {
      "title": "HTML5",
    },
    {
      "title": "jQuery",
    },
    {
      "title": "XML",
    },
    {
      "title": "JSON",
    },
    {
      "title": "SQL",
    },
    {
      "title": "AJAX",
    },
    {
      "title": "REST",
    },
    {
      "title": "HTML 5",
    },
    {
      "title": "Backbone.js",
    },
    {
      "title": "CSS",
    },
    {
      "title": "Linux",
    },
    {
      "title": "CSS3",
    },
    {
      "title": "PostgreSQL",
    },
    {
      "title": "Node.js",
    },
  ],
  "volunteerExperience": [],
}
`;

exports[`should get profiles properly 2`] = `
{
  "about": {
    "text": "Quoi de mieux qu'un sp√©cialiste de la tech¬†üåê pour recruter un profil tech, un sp√©cialiste en Ressources humaines pour recruter un RH. Logique non ?¬†üòÅ Je ne me contente pas de remplir des cases et proposer des profils. Je suis avant tout √† l‚Äô√©coute de tout challenge en mati√®re de staffing. Freelances, on vous propose des missions, vous disposez. D√©cisionnaires de la tech et du digital, pas besoin de vous connecter, de d√©finir vos recherches de comp√©tences, on le fait pour vous, on vous propose des profils ad‚Äôhoc, vite. A tous, nous g√©rons la mission de la fa√ßon la plus humaine possible, un suivi hebdomadaire et une marge transparente.",
  },
  "accomplishments": [],
  "contact": [
    {
      "links": [
        "https://www.linkedin.com/in/lina-morel-9b1b7b1b8",
      ],
      "type": "Profil de Lina",
      "values": [
        "linkedin.com/in/lina-morel-9b1b7b1b8",
      ],
    },
    {
      "links": [
        "mailto:lina.morel@bloomays.com",
      ],
      "type": "E-mail",
      "values": [
        "lina.morel@bloomays.com",
      ],
    },
    {
      "type": "Connect√©",
      "values": [
        "13 nov. 2020",
      ],
    },
  ],
  "courses": [],
  "educations": [
    {
      "date1": "2016",
      "date2": "2019",
      "degree": "Programme grande √©cole, Marketing",
      "description": "The mission of the BIB program is to provide students with a broad introduction to management, international business, foreign cultures and foreign languages which will allow them to work in a diverse environment or pursue further graduate management studies.",
      "skills": [
        {
          "title": "Marketing digital",
        },
        {
          "title": "Growth hacking",
        },
        {
          "title": "Marketing",
        },
      ],
      "title": "Grenoble Ecole de Management",
      "url": "https://www.linkedin.com/company/18927/",
    },
    {
      "date1": "2014",
      "date2": "2016",
      "degree": "BTS NRC (N√©gociation et Relation Client), Commerce international / commerce",
      "title": "Lyc√©e Emile Dubois",
      "url": "https://www.linkedin.com/search/results/all/?keywords=Lyc%C3%A9e+Emile+Dubois",
    },
  ],
  "languages": [],
  "peopleAlsoViewed": [],
  "positions": [
    {
      "companyName": "Bloomays",
      "contractType": "CDI",
      "date1": "mai 2020",
      "date2": "aujourd‚Äôhui",
      "description": "Bloomays c‚Äôest une start-up exclusivement d√©di√©e √† l‚Äôaccompagnement RH des CTO !
On r√©sout tous vos probl√®mes/challenges li√©s aux √©quipes: recrutement (interne ou prestation externe), √©valuation de ces √©quipes √† r√©ussir leurs projets. Motivation et r√©tention, mobilit√©s internes, investissement dans les comp√©tences cl√©s/critiques.
CTO, on fait tout pour te faciliter la vie !",
      "link": "https://www.linkedin.com/company/14823353/",
      "location": "Ville de Paris, √éle-de-France, France",
      "remotePolicy": "Hybride",
      "roles": [],
      "skills": [
        {
          "title": "Growth hacking",
        },
      ],
      "title": "Growth Hacker",
      "url": "https://www.linkedin.com/company/14823353/",
    },
    {
      "companyName": "Brentt",
      "companyNameWithRole": undefined,
      "link": "https://www.linkedin.com/company/28896199/",
      "roles": [
        {
          "date1": "juin 2019",
          "date2": "f√©vr. 2020",
          "location": "Paris, √éle-de-France, France",
          "remotePolicy": "Sur site",
          "title": "Responsable d√©veloppement commercial marketing",
        },
        {
          "date1": "f√©vr. 2019",
          "date2": "mai 2019",
          "description": "Stage de fin d'√©tudes.",
          "location": "Paris, √éle-de-France, France",
          "remotePolicy": "Hybride",
          "skills": [
            {
              "title": "Marketing",
            },
          ],
          "title": "Assistant responsable marketing",
        },
      ],
      "url": "https://www.linkedin.com/company/28896199/",
    },
  ],
  "profile": {
    "connections": "Plus de 500¬†relations",
    "headline": "Growth Hacker & Tech Recruiter chez Bloomays",
    "imageurl": Any<String>,
    "location": "Paris, √éle-de-France, France",
    "name": "Lina Morel",
    "summary": "Quoi de mieux qu'un sp√©cialiste de la tech¬†üåê pour recruter un profil tech, un sp√©cialiste en Ressources humaines pour recruter un RH. Logique non ?¬†üòÅ Je ne me contente pas de remplir des cases et proposer des profils. Je suis avant tout √† l‚Äô√©coute de tout challenge en mati√®re de staffing. Freelances, on vous propose des missions, vous disposez. D√©cisionnaires de la tech et du digital, pas besoin de vous connecter, de d√©finir vos recherches de comp√©tences, on le fait pour vous, on vous propose des profils ad‚Äôhoc, vite. A tous, nous g√©rons la mission de la fa√ßon la plus humaine possible, un suivi hebdomadaire et une marge transparente.",
  },
  "projects": [],
  "recommendations": {
    "given": [],
    "givenCount": "0",
    "received": [],
    "receivedCount": "0",
  },
  "skills": [
    {
      "title": "D√©veloppement commercial",
    },
    {
      "title": "Marketing digital",
    },
    {
      "title": "Growth hacking",
    },
    {
      "title": "Test¬†A/B",
    },
    {
      "title": "Marketing",
    },
    {
      "title": "Optimisation des conversations",
    },
    {
      "title": "Email marketing",
    },
    {
      "title": "Optimisation pour les moteurs de recherche (SEO)",
    },
    {
      "title": "Acquisition de clients",
    },
    {
      "title": "Web analytique",
    },
    {
      "title": "anglais",
    },
    {
      "title": "Recrutement",
    },
    {
      "title": "talent acquisition",
    },
    {
      "title": "D√©veloppement commercial",
    },
    {
      "title": "Marketing digital",
    },
    {
      "title": "Growth hacking",
    },
    {
      "title": "Test¬†A/B",
    },
    {
      "title": "Marketing",
    },
    {
      "title": "Email marketing",
    },
    {
      "title": "Optimisation pour les moteurs de recherche (SEO)",
    },
    {
      "title": "Acquisition de clients",
    },
    {
      "title": "Web analytique",
    },
    {
      "title": "Recrutement",
    },
    {
      "title": "talent acquisition",
    },
    {
      "title": "anglais",
    },
  ],
  "volunteerExperience": [],
}
`;

exports[`should get profiles properly 3`] = `
{
  "about": {
    "text": "Enchant√©e ! Une üë©üèª exceptionnelle üî•, l√† pour te permettre de trouver ton job TECH/RH id√©al ü§© parmi les meilleures opportunit√©s du march√© au sein de startups, scale-ups, d'√©diteurs de logiciels ou de grands comptes. J'ai r√©guli√®rement de nouvelles offres √† proposer alors n'h√©sites pas √† m'ajouter √† ton r√©seau et √† me contacter. üòâ Mon but c'est de te proposer la bonne mission, au bon moment ! Pour √ßa, pas de secret ü§´, je m'int√©resse √† ton profil et t'envoie un message sp√©cialement destin√©. N'h√©site pas √† prendre contact avec moi. √Ä tr√®s vite dans la communaut√© des Bloomers üë®üèΩ‚Äçüíªüë©üèª‚Äçüíª",
  },
  "accomplishments": [],
  "contact": [
    {
      "links": [
        "https://www.linkedin.com/in/rosa-barbet-69a6bb220",
      ],
      "type": "Profil de Rosa",
      "values": [
        "linkedin.com/in/rosa-barbet-69a6bb220",
      ],
    },
    {
      "links": [
        "mailto:barbetrosa@gmail.com",
      ],
      "type": "E-mail",
      "values": [
        "barbetrosa@gmail.com",
      ],
    },
    {
      "type": "Connect√©",
      "values": [
        "13 oct. 2021",
      ],
    },
  ],
  "courses": [],
  "educations": [
    {
      "date1": "sept. 2012",
      "date2": "ao√ªt 2015",
      "degree": "Dipl√¥me d'√©tudes sup√©rieures en management, Management",
      "title": "Grenoble Ecole de Management",
      "url": "https://www.linkedin.com/company/18927/",
    },
    {
      "date1": "sept. 2011",
      "date2": "juin 2012",
      "degree": "DUT gestion des entreprises et des administrations, Gestion des ressources humaines / administration du personnel, g√©n√©ral",
      "title": "IUT de Paris - Rives de Seine",
      "url": "https://www.linkedin.com/company/15096879/",
    },
  ],
  "languages": [],
  "peopleAlsoViewed": [],
  "positions": [
    {
      "companyName": "Bloomays",
      "contractType": "CDI",
      "date1": "juil. 2021",
      "date2": "aujourd‚Äôhui",
      "link": "https://www.linkedin.com/company/14823353/",
      "title": "Talent and Account Manager (TAM)",
      "url": "https://www.linkedin.com/company/14823353/",
    },
    {
      "companyName": "Havas",
      "contractType": "CDI",
      "date1": "ao√ªt 2018",
      "date2": "mai 2021",
      "link": "https://www.linkedin.com/company/2500/",
      "location": "Ville de Paris, √éle-de-France, France",
      "title": "Recruteuse technique",
      "url": "https://www.linkedin.com/company/2500/",
    },
    {
      "companyName": "HP",
      "contractType": "CDI",
      "date1": "janv. 2016",
      "date2": "mai 2018",
      "link": "https://www.linkedin.com/company/5390798/",
      "location": "Grenoble, Auvergne-Rh√¥ne-Alpes, France",
      "title": "Charg√©e de Recrutement",
      "url": "https://www.linkedin.com/company/5390798/",
    },
    {
      "companyName": "Melloon",
      "contractType": "Stage",
      "date1": "juil. 2015",
      "date2": "janv. 2016",
      "link": "https://www.linkedin.com/search/results/all/?keywords=Melloon",
      "location": "Madrid, Communaut√© de Madrid, Espagne",
      "title": "Assistante responsable ressources humaines",
      "url": "https://www.linkedin.com/search/results/all/?keywords=Melloon",
    },
  ],
  "profile": {
    "connections": "Plus de 500¬†relations",
    "headline": "Talent & Account Manager (TAM)",
    "imageurl": Any<String>,
    "location": "Paris, √éle-de-France, France",
    "name": "Rosa Barbet",
    "summary": "Enchant√©e ! Une üë©üèª exceptionnelle üî•, l√† pour te permettre de trouver ton job TECH/RH id√©al ü§© parmi les meilleures opportunit√©s du march√© au sein de startups, scale-ups, d'√©diteurs de logiciels ou de grands comptes. J'ai r√©guli√®rement de nouvelles offres √† proposer alors n'h√©sites pas √† m'ajouter √† ton r√©seau et √† me contacter. üòâ Mon but c'est de te proposer la bonne mission, au bon moment ! Pour √ßa, pas de secret ü§´, je m'int√©resse √† ton profil et t'envoie un message sp√©cialement destin√©. N'h√©site pas √† prendre contact avec moi. √Ä tr√®s vite dans la communaut√© des Bloomers üë®üèΩ‚Äçüíªüë©üèª‚Äçüíª",
  },
  "projects": [],
  "recommendations": {
    "given": [],
    "givenCount": "0",
    "received": [],
    "receivedCount": "0",
  },
  "skills": [
    {
      "title": "Droit",
    },
    {
      "title": "Ressources humaines (RH)",
    },
    {
      "title": "Sourcing",
    },
    {
      "title": "Recrutement",
    },
    {
      "title": "Recrutement technique",
    },
    {
      "title": "Strat√©gie de sourcing",
    },
    {
      "title": "Droit",
    },
    {
      "title": "Ressources humaines (RH)",
    },
    {
      "title": "Sourcing",
    },
    {
      "title": "Recrutement",
    },
    {
      "title": "Recrutement technique",
    },
    {
      "title": "Strat√©gie de sourcing",
    },
  ],
  "volunteerExperience": [],
}
`;

exports[`should get profiles properly 4`] = `
{
  "about": {
    "text": "Hello ! Une recruteuse de feu üë©üèª, l√† pour t'aider √† trouver ton job TECH id√©al ü§© parmi les meilleures opportunit√©s du march√© au sein de startups, d'√©diteurs de logiciels ou de grands comptes. J'ai r√©guli√®rement de nouvelles offres exclusives √† proposer alors n'h√©sites pas √† m'ajouter √† ton r√©seau ou √† me contacter. üòâ Mon but c'est de te proposer la bonne mission, au parfait moment ! üïõ Pour √ßa, pas de secret ü§´, en s'int√©ressant √† ton profil et en t'envoyant un message sp√©cialement destin√©. √Ä tr√®s vite dans la communaut√© des Bloomers üë®üèΩ‚Äçüíªüë©üèª‚Äçüíª",
  },
  "accomplishments": [],
  "contact": [
    {
      "links": [
        "https://www.linkedin.com/in/aurore-veneto-5a6034213",
      ],
      "type": "Profil de Aurore",
      "values": [
        "linkedin.com/in/aurore-veneto-5a6034213",
      ],
    },
    {
      "links": [
        "mailto:aurore.veneto@gmail.com",
      ],
      "type": "E-mail",
      "values": [
        "aurore.veneto@gmail.com",
      ],
    },
    {
      "type": "Connect√©",
      "values": [
        "21 juin 2021",
      ],
    },
  ],
  "courses": [],
  "educations": [
    {
      "date1": "2014",
      "date2": "2017",
      "degree": "Master en Ressources Humaines, Gestion des ressources humaines / administration du personnel, g√©n√©ral",
      "title": "Rennes School of Business",
      "url": "https://www.linkedin.com/company/15092681/",
    },
    {
      "date1": "2013",
      "date2": "2014",
      "degree": "BTS NRC, N√©gociations & Relations clients",
      "title": "ISCG Paris",
      "url": "https://www.linkedin.com/company/2090001/",
    },
  ],
  "languages": [],
  "peopleAlsoViewed": [],
  "positions": [
    {
      "companyName": "Bloomays",
      "contractType": "Temps plein",
      "date1": "nov. 2020",
      "date2": "aujourd‚Äôhui",
      "description": "En tant que TAM, j'accompagne mes clients sur le recrutement de leurs √©quipes !

üëâ Tous types de profils : Software Engineer, Tech Lead, CTO, DevOps, QA Engineer, Technical Architect, Product Owner, Product Manager, UX/UI/Product Designers...
üëâ Tous types de technos : JavaScript, Java, PHP, Python, RoR, Go, mobile, blockchain...",
      "link": "https://www.linkedin.com/company/14823353/",
      "location": "Ville de Paris, √éle-de-France, France",
      "roles": [],
      "title": "Talent & Account Manager",
      "url": "https://www.linkedin.com/company/14823353/",
    },
    {
      "companyName": "Sopra Steria",
      "contractType": "Temps plein",
      "date1": "juil. 2018",
      "date2": "ao√ªt 2020",
      "description": "- Sourcing on job-boards, pre-qualifications and physical interview
- Integration of the candidates and tracking
- Clients follow-up, prospection and maintain of the portfolio
- Push strategy and commercial proposals
- Creation of partnerships towards schools and training institution",
      "link": "https://www.linkedin.com/company/3054/",
      "location": "Ville de Paris, √éle-de-France, France",
      "roles": [],
      "title": "Tech Recruiter",
      "url": "https://www.linkedin.com/company/3054/",
    },
    {
      "companyName": "Capgemini",
      "contractType": "CDD",
      "date1": "sept. 2017",
      "date2": "mai 2018",
      "link": "https://www.linkedin.com/company/157240/",
      "location": "Ville de Paris, √éle-de-France, France",
      "title": "Charg√©e de Recrutement",
      "url": "https://www.linkedin.com/company/157240/",
    },
  ],
  "profile": {
    "connections": "Plus de 500¬†relations",
    "headline": "Talent & Account Manager (TAM) chez Bloomays",
    "imageurl": Any<String>,
    "location": "Paris, √éle-de-France, France",
    "name": "Aurore Veneto",
    "summary": "Hello ! Une recruteuse de feu üë©üèª, l√† pour t'aider √† trouver ton job TECH id√©al ü§© parmi les meilleures opportunit√©s du march√© au sein de startups, d'√©diteurs de logiciels ou de grands comptes. J'ai r√©guli√®rement de nouvelles offres exclusives √† proposer alors n'h√©sites pas √† m'ajouter √† ton r√©seau ou √† me contacter. üòâ Mon but c'est de te proposer la bonne mission, au parfait moment ! üïõ Pour √ßa, pas de secret ü§´, en s'int√©ressant √† ton profil et en t'envoyant un message sp√©cialement destin√©. √Ä tr√®s vite dans la communaut√© des Bloomers üë®üèΩ‚Äçüíªüë©üèª‚Äçüíª",
  },
  "projects": [],
  "recommendations": {
    "given": [],
    "givenCount": "0",
    "received": [],
    "receivedCount": "0",
  },
  "skills": [
    {
      "title": "Anglais",
    },
    {
      "title": "Recherche",
    },
    {
      "title": "Marketing",
    },
    {
      "title": "Formation",
    },
    {
      "title": "Comp√©tences analytiques",
    },
    {
      "title": "Syst√®me de suivi des candidatures",
    },
    {
      "title": "Sourcing",
    },
    {
      "title": "Strat√©gie de sourcing",
    },
    {
      "title": "Entretiens",
    },
    {
      "title": "Entretien sur la base des comp√©tences",
    },
    {
      "title": "Marketing",
    },
    {
      "title": "Formation",
    },
    {
      "title": "Sourcing",
    },
    {
      "title": "Strat√©gie de sourcing",
    },
    {
      "title": "Entretiens",
    },
    {
      "title": "Entretien sur la base des comp√©tences",
    },
    {
      "title": "Syst√®me de suivi des candidatures",
    },
    {
      "title": "Recherche",
    },
    {
      "title": "Comp√©tences analytiques",
    },
    {
      "title": "Anglais",
    },
  ],
  "volunteerExperience": [],
}
`;
